### PR TITLE
Fix module_utils.common.test_collections testsuite

### PR DESCRIPTION
**Pull request by webknjaz**
##### SUMMARY

Apparently, the test file was named incorrectly.
Thanks to @Andersson007 for pointing this out.

UPD: 2 tests are failing under Python 2.7.
```console
...
[gw0] [ 93%] FAILED test/units/module_utils/common/test_collections.py::test_iterable_positive[seq5]
...
[gw7] [100%] FAILED test/units/module_utils/common/test_collections.py::TestImmutableDict::test_repr

=================================== FAILURES ===================================
_________________________ test_iterable_positive[seq5] _________________________
[gw0] linux2 -- Python 2.7.12 /tmp/python-BJx475-ansible/python
seq = <units.module_utils.common.test_collections.IterableStub object at 0x7f2989dfea90>

    @pytest.mark.parametrize(
        'seq',
        ([], (), {}, set(), frozenset(), IterableStub()),
    )
    def test_iterable_positive(seq):
>       assert is_iterable(seq)
E       assert False
E        +  where False = is_iterable(<units.module_utils.common.test_collections.IterableStub object at 0x7f2989dfea90>)

test/units/module_utils/common/test_collections.py:85: AssertionError
_________________________ TestImmutableDict.test_repr __________________________
[gw7] linux2 -- Python 2.7.12 /tmp/python-BJx475-ansible/python
self = <units.module_utils.common.test_collections.TestImmutableDict object at 0x7f2805273c50>

    def test_repr(self):
        imdict = ImmutableDict({1: 2, 'a': 'b'})
>       assert repr(imdict) == "ImmutableDict({1: 2, 'a': 'b'})"
E       assert "ImmutableDic...: 'b', 1: 2})" == "ImmutableDict...2, 'a': 'b'})"
E         - ImmutableDict({'a': 'b', 1: 2})
E         ?                        ------
E         + ImmutableDict({1: 2, 'a': 'b'})
E         ?                ++++++

test/units/module_utils/common/test_collections.py:156: AssertionError
```

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

N/A

##### ADDITIONAL INFORMATION

N/A

### CODE REVIEW

**Comment by abadger** [comment]
The ImmutableDict problem is a problem in the test.  It's assuming that the sort order is predictable but it isn't.  /me looks for a good solution

**Comment by webknjaz** [comment]
@abadger I've fixed that

File: test/units/module_utils/common/test_collections.py [hunk 1]
```diff
@@ -126,13 +127,13 @@ def test_from_kwargs(self):
     def test_immutable(self):
         imdict = ImmutableDict({1: 2})
 
-        with pytest.raises(TypeError) as exc_info:
+        expected_reason = r"^'ImmutableDict' object does not support item assignment$"
+
+        with pytest.raises(TypeError, match=expected_reason):
```
Comment (abadger) [inline test/units/module_utils/common/test_collections.py:132]: Be sure that this fails in shippable on error.  For a while we were pinned on a version of pytest where match did not work.

Comment (webknjaz) [reply test/units/module_utils/common/test_collections.py:132]: Well, master seems green + I've checked it locally in docker containers. I assume that pytest should be the same...

Comment (abadger) [reply test/units/module_utils/common/test_collections.py:132]: I mean, you need to test that it properly fails if you raise the correct exception but with the wrong message.  The last time I checked in shippable (but could have been as much as a year ago) pytest's match parameter didn't work.  If I raised ```TypeError('Incorrect message')``` pytest.raises would have thought it was fine.

Comment (abadger) [reply test/units/module_utils/common/test_collections.py:132]: (It correctly fails locally, so I am pretty sure that the latest pytest doesn't have the problem.  But I don't know if we're still pinning an older version of pytest in shippable which will not handle match right)

Comment (webknjaz) [reply test/units/module_utils/common/test_collections.py:132]: It's `pytest-4.4.0` according to https://app.shippable.com/github/ansible/ansible/runs/125466/10/console

**Pull request merged**